numba/core/event.py

```diff
@@ -423,3 +426,42 @@ def on_exit(*exc_details):
 
         start_event(kind, data=data)
         yield
+
+
+def _write_chrome_trace(rec):
+    """ If chrome tracing is enabled,
+        write event information (process/thread id, event name,
+        and duration time to json file.
+    """
+    import os
+    import threading
+
+    pid = os.getpid()
+    tid = threading.get_ident()
+    evs = []
+    for ts, rec in rec.buffer:
+        data = rec.data
+        cat = str(rec.kind)
+        ph = 'B' if rec.is_start else 'E'
+        pid = pid
+        tid = tid
+        ts = ts
+        name = data['name']
+        args = data
+        ev = dict(
+            cat=cat, pid=pid, tid=tid, ts=ts, ph=ph, name=name, args=args,
+        )
+        evs.append(ev)
+    return evs
+
+
+if config.CHROME_TRACE:
+    listener = RecordingListener()
+    register("nb:run_pass", listener)
+    filename = config.CHROME_TRACE
```

Comment:
I don't think this is going to be multiprocessing safe? Should the filename be combined with the `pid` of the parent process.

Reply:
Added warning in docs about the limitation for multiprocessing